core/src/main/java/jenkins/telemetry/Telemetry.java

```diff
@@ -0,0 +1,196 @@
+
+    // https://webhook.site is a nice stand-in for this during development; just needs to end in ? to submit the ID as query parameter
+    @Restricted(NoExternalUse.class)
+    @VisibleForTesting
+    static String ENDPOINT = SystemProperties.getString(Telemetry.class.getName() + ".endpoint", "https://uplink.jenkins.io/events");
+
+    private static final Logger LOGGER = Logger.getLogger(Telemetry.class.getName());
+
+    /**
+     * ID of this collector, typically a basic alphanumeric string (and _- characters).
+     *
+     * Good IDs are globally unique and human readable (i.e. no UUIDs).
+     *
+     * For a periodically updated list of all public implementations, see https://jenkins.io/doc/developer/extensions/jenkins-core/#telemetry
+     * 
+     * @return ID of the collector, never null or empty
+     */
+    @Nonnull
+    public abstract String getId();
+
+    /**
+     * User friendly display name for this telemetry collector, ideally localized.
+     *
+     * @return display name, never null or empty
+     */
+    @Nonnull
+    public abstract String getDisplayName();
+
+    /**
+     * Start date for the collection.
+     * Will be checked in Jenkins to not collect outside the defined time span.
+     * This does not have to be precise enough for time zones to be a consideration.
+     *
+     * @return collection start date
+     */
+    @Nonnull
+    public abstract LocalDate getStart();
+
+    /**
+     * End date for the collection.
+     * Will be checked in Jenkins to not collect outside the defined time span.
+     * This does not have to be precise enough for time zones to be a consideration.
+     *
+     * @return collection end date
+     */
+    @Nonnull
+    public abstract LocalDate getEnd();
+
+    /**
+     * Returns the content to be sent to the telemetry service.
+     *
+     * This method is called periodically, once per content submission.
+     *
+     * @return
+     */
+    @Nonnull
+    public abstract JSONObject createContent();
+
+    public static ExtensionList<Telemetry> all() {
+        return ExtensionList.lookup(Telemetry.class);
+    }
+
+    @Extension
+    public static class TelemetryReporter extends AsyncPeriodicWork {
+
+        public TelemetryReporter() {
+            super("telemetry collection");
+        }
+
+        @Override
+        public long getRecurrencePeriod() {
+            return TimeUnit.HOURS.toMillis(24);
+        }
+
+        @Override
+        protected void execute(TaskListener listener) throws IOException, InterruptedException {
+            if (UsageStatistics.DISABLED || !Jenkins.getInstance().isUsageStatisticsCollected()) {
+                LOGGER.info("Collection of anonymous usage statistics is disabled, skipping telemetry collection and submission");
+                return;
+            }
+            Telemetry.all().forEach(telemetry -> {
+                if (telemetry.getStart().isAfter(LocalDate.now())) {
```

Comment:
Note to self: time zone is ignored here as per Javadoc on `getStart` / `getEnd`, so collectors may effectively start/end incrementally over the course of a day.